shared/js/project.js

```diff
@@ -190,7 +200,17 @@ module.exports = {
 
     current = data;
     if (isNewChannel) {
-      location.href = current.level + '#' + current.id + '/edit';
+      // We have a new channel, meaning either we had no channel before, or
+      // we've changed channels.
+
+      if (location.hash || !window.history.pushState) {
+        // We're using a hash route or don't support replace state. Use our hash
+        // based route to ensure we don't have a page load.
+        location.href = current.level + '#' + current.id + '/edit';
+      } else {
+        window.history.pushState(null, document.title,
```

Comment:
when we do pushState, the browser doesn't know about the previous page, and "back" behavior (if desired) has to be implemented by handling the popstate event. However, I don't think we want to support going back to the previous URL in this case, so we should be ok.